### PR TITLE
Simplify the Hero form and add basic styling

### PR DESCRIPTION
**Pull request by mortenson**
This PR is ready to be looked at, but I'd like input from @briancwald about what styling needs to be supported before committing this work. So far the only changes are:

1. Remove the fieldsets from the form, and move advanced elements like the nested block and icon to the "Advanced" section.
2. Do a lot of form cleanup to simplify elements, including patching Entity Browser to not use a table for displaying single-cardinality Image fields.
3. Added a field to change alignment, which required adding some theme-agnostic CSS to df_tools_block.
4. Added a field to change text color, using the browser's built in Color picker.

We can go pretty crazy with styles, but some that may be useful to add are:

1. Gradient/background color - take the white gradient out of the Image Style for heroes and put it into CSS like it was in Drupal 7, and let users change the gradient color.
2. Callout link color - this could also be controlled globally in colors.css, so instead of doing a free-form color field we could follow Foundation's standard and just allow success, secondary, alert, and info.
3. Font size and family - I'm against this but the feature has been requested multiple times.

Any thoughts?

### CODE REVIEW

**Comment by briancwald** [comment]
> We can go pretty crazy with styles, but some that may be useful to add are:

Agree on 1 and 2. 

Point 3, we should allow font size changes, but NOT font family.

**Comment by mortenson** [comment]
I've implemented callout and gradient color, but not font size changes. Thinking about it more, I would rather just get what we had in D7 in, then add more styles as they're requested. Plus configuring font size (by pixel) has issues with mobile support, and as a user I felt weird not being able to change the font style and weight as well.

File: modules/df/df_tools/df_tools_blocks/css/hero.css [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  color: white;
+  border: 1px solid white;
+  background: #2199e8;
+  padding: 10px;
+  filter: brightness(100%);
+  transition: .2s;
+}
+
+.hero-block .field-name-field-hero-link a:hover {
+  filter: brightness(85%);
```
Comment (mortenson) [inline modules/df/df_tools/df_tools_blocks/css/hero.css:43]: This is a creative way to generically apply hover styles to colors without manually writing in rules. This way the user can set the background to anything, and we calculate a darker hover color for it.

File: modules/df/df_tools/df_tools_blocks/css/hero.css [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+.hero-block .field-name-field-hero-image img {
+  width: 100%;
+}
+
+.hero-block .field-name-field-hero-image:before {
+  content: '';
+  position: absolute;
+  background-image: inherit;
```
Comment (mortenson) [inline modules/df/df_tools/df_tools_blocks/css/hero.css:61]: You can't define styles inline for pseudo elements, so PHP sets an inline style for .field-name-field-hero-image, and this inherits it. Another thing learned while making this PR.

**Comment by briancwald** [comment]
Functionally this works well! the UX and design needs a fair bit of work before we can merge. I will focus on this when I can.

**Comment by saltednut** [comment]
I think we can commit this once the merge conflicts are resolved.

**Pull request merged**